community/ndp/messaging-v1/src/main/java/org/neo4j/ndp/messaging/v1/infrastructure/ValueUnboundRelationship.java

```diff
@@ -95,6 +93,26 @@ public long getId()
     }
 
     @Override
+    public Map<String, Object> getProperties( String... names )
+    {
+        Map<String, Object> selectedProperties = new HashMap<>();
+        for ( String name : names )
+        {
+            if ( props.containsKey( name ) )
+            {
+                selectedProperties.put( name, props.get( name ) );
+            }
+        }
+        return selectedProperties;
+    }
+
+    @Override
+    public Map<String, Object> getAllProperties()
+    {
+        return props;
+    }
+
+    @Override
```

Comment:
This code has been duplicate 3 times, please let's try to avoid so much duplication.